zebra-chain/src/amount.rs

```diff
@@ -701,4 +710,56 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_sum() -> Result<()> {
+        zebra_test::init();
+
+        let one: Amount = 1.try_into()?;
+        let neg_one: Amount = (-1).try_into()?;
+
+        let zero: Amount = 0.try_into()?;
+
+        // success
+        let amounts = vec![one, neg_one, zero];
+        let sum: Amount = amounts.iter().copied().sum::<Result<Amount, Error>>()?;
```

Comment:
Do you also want to `impl<C> std::iter::Sum<&Amount<C>> for Result<Amount<C>>`?
That way, you don't have to add `.copied()` all the time.

This change is totally optional.

Reply:
I think this is a good idea, i actually tried to do it but had compiler problems with the reference lifetimes that i was not able to resolve. Will need some assistance there.

Reply:
I opened PR #2506, you're right, it was a bit tricky.

Reply:
Thanks, that was not an easy job. I merged your branch into this one.